diff --git a/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/builder/widget/BaseWidget.java b/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/builder/widget/BaseWidget.java
index 2c70de7..f587766 100644
--- a/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/builder/widget/BaseWidget.java
+++ b/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/builder/widget/BaseWidget.java
@@ -9,8 +9,6 @@ import android.widget.LinearLayout;
 
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReadableMap;
-import com.facebook.react.uimanager.Spacing;
-import com.facebook.react.views.view.ReactViewBackgroundDrawable;
 
 import java.util.Collections;
 import java.util.HashMap;
@@ -24,7 +22,10 @@ public abstract class BaseWidget<T extends View> {
 
     protected T view;
 
-    private ReactViewBackgroundDrawable mReactViewBackground;
+    private GradientDrawable mReactViewBackground;
+    private int mBorderColor = Color.TRANSPARENT;
+    private int mBorderWidth = 0;
+    private final float[] mBorderRadii = new float[8];
 
     public BaseWidget(ReactApplicationContext context, ReadableMap props, List<View> children) {
         this.appContext = context;
@@ -126,46 +127,30 @@ public abstract class BaseWidget<T extends View> {
     private void setBorder() {
         if (props.hasKey("borderColor")) {
             ReadableMap borderColor = props.getMap("borderColor");
-            getReactViewBackground().setBorderColor(
-                Spacing.LEFT,
-                Color.parseColor(borderColor.getString("left")),
-                Color.alpha(Color.parseColor(borderColor.getString("left")))
-            );
-            getReactViewBackground().setBorderColor(
-                Spacing.RIGHT,
-                Color.parseColor(borderColor.getString("right")),
-                Color.alpha(Color.parseColor(borderColor.getString("right")))
-            );
-            getReactViewBackground().setBorderColor(
-                Spacing.TOP,
-                Color.parseColor(borderColor.getString("top")),
-                Color.alpha(Color.parseColor(borderColor.getString("top")))
-            );
-            getReactViewBackground().setBorderColor(
-                Spacing.BOTTOM,
-                Color.parseColor(borderColor.getString("bottom")),
-                Color.alpha(Color.parseColor(borderColor.getString("bottom")))
-            );
+            mBorderColor = Color.parseColor(borderColor.getString("left"));
+            updateBorder();
         }
 
         if (props.hasKey("borderWidth")) {
             ReadableMap borderWidth = props.getMap("borderWidth");
-            getReactViewBackground().setBorderWidth(Spacing.LEFT, dpToPx(borderWidth.getDouble("left")));
-            getReactViewBackground().setBorderWidth(Spacing.RIGHT, dpToPx(borderWidth.getDouble("right")));
-            getReactViewBackground().setBorderWidth(Spacing.TOP, dpToPx(borderWidth.getDouble("top")));
-            getReactViewBackground().setBorderWidth(Spacing.BOTTOM, dpToPx(borderWidth.getDouble("bottom")));
+            mBorderWidth = Math.max(
+                Math.max(dpToPx(borderWidth.getDouble("left")), dpToPx(borderWidth.getDouble("right"))),
+                Math.max(dpToPx(borderWidth.getDouble("top")), dpToPx(borderWidth.getDouble("bottom")))
+            );
+            updateBorder();
         }
 
         if (props.hasKey("borderRadius")) {
             ReadableMap borderRadius = props.getMap("borderRadius");
-            getReactViewBackground().setRadius(dpToPx(borderRadius.getDouble("topRight")), 1);
-            getReactViewBackground().setRadius(dpToPx(borderRadius.getDouble("bottomRight")), 2);
-            getReactViewBackground().setRadius(dpToPx(borderRadius.getDouble("bottomLeft")), 3);
-            getReactViewBackground().setRadius(dpToPx(borderRadius.getDouble("topLeft")), 0);
-        }
-
-        if (props.hasKey("borderStyle")) {
-            getReactViewBackground().setBorderStyle(props.getString("borderStyle").toUpperCase());
+            mBorderRadii[0] = dpToPx(borderRadius.getDouble("topLeft"));
+            mBorderRadii[1] = dpToPx(borderRadius.getDouble("topLeft"));
+            mBorderRadii[2] = dpToPx(borderRadius.getDouble("topRight"));
+            mBorderRadii[3] = dpToPx(borderRadius.getDouble("topRight"));
+            mBorderRadii[4] = dpToPx(borderRadius.getDouble("bottomRight"));
+            mBorderRadii[5] = dpToPx(borderRadius.getDouble("bottomRight"));
+            mBorderRadii[6] = dpToPx(borderRadius.getDouble("bottomLeft"));
+            mBorderRadii[7] = dpToPx(borderRadius.getDouble("bottomLeft"));
+            getReactViewBackground().setCornerRadii(mBorderRadii);
         }
     }
 
@@ -207,15 +192,19 @@ public abstract class BaseWidget<T extends View> {
         }
     }
 
-    private ReactViewBackgroundDrawable getReactViewBackground() {
+    private GradientDrawable getReactViewBackground() {
         if (mReactViewBackground == null) {
-            mReactViewBackground = new ReactViewBackgroundDrawable(view.getContext());
+            mReactViewBackground = new GradientDrawable();
             view.setBackground(null);
             view.setBackground(mReactViewBackground);
         }
         return mReactViewBackground;
     }
 
+    private void updateBorder() {
+        getReactViewBackground().setStroke(mBorderWidth, mBorderColor);
+    }
+
     private Map<String, Object> getObjectOrEmptyMap(ReadableMap propMap) {
         HashMap<String, Object> objectMap;
         if (propMap != null) {
